### PR TITLE
Add new pango attributes

### PR DESCRIPTION
**Pull request by sophie-h**

### CODE REVIEW

File: pango/src/attribute.rs [hunk 1]
```diff
@@ -14,96 +20,153 @@ use crate::Weight;
...
+    #[cfg(any(feature = "v1_44", feature = "dox"))]
+    #[cfg_attr(feature = "dox", doc(cfg(feature = "v1_44")))]
+    #[doc(alias = "pango_attr_allow_breaks_new")]
+    pub fn new_allow_breaks(allow_breaks: bool) -> Option<Attribute> {
+        unsafe { from_glib_full(ffi::pango_attr_allow_breaks_new(allow_breaks.to_glib())) }
+    }
+
     #[cfg(any(feature = "v1_38", feature = "dox"))]
     #[cfg_attr(feature = "dox", doc(cfg(feature = "v1_38")))]
+    #[doc(alias = "pango_attr_background_alpha_new")]
```
Comment (sdroege) [inline pango/src/attribute.rs:32]: @GuillaumeGomez Why didn't your script add those?

File: pango/src/attribute.rs [hunk 2]
```diff
@@ -14,96 +20,153 @@ use crate::Weight;
...
+    #[cfg_attr(feature = "dox", doc(cfg(feature = "v1_46")))]
+    #[doc(alias = "pango_attr_overline_color_new")]
+    pub fn new_overline_color(red: u16, green: u16, blue: u16) -> Option<Attribute> {
+        unsafe { from_glib_full(ffi::pango_attr_overline_color_new(red, green, blue)) }
+    }
+
+    #[cfg(any(feature = "v1_46", feature = "dox"))]
+    #[cfg_attr(feature = "dox", doc(cfg(feature = "v1_46")))]
+    #[doc(alias = "pango_attr_overline_new")]
+    pub fn new_overline(overline: Overline) -> Option<Attribute> {
```
Comment (sdroege) [inline pango/src/attribute.rs:103]: Are all these constructors actually `nullable`? That seems strange for the majority of them

**Review by sdroege — COMMENTED** [review]
Also it seems like the `new_` prefix of all the constructors is a bit un-Rust-y and should probably be removed.

**Comment by fengalin** [comment]
> Also it seems like the `new_` prefix of all the constructors is a bit un-Rust-y and should probably be removed.

What do you mean? Most Rust constructors use `new` AFAICT :) The problem here is that we construct a variant of the type. If we'd used `overline` instead of `new_overline` that would potentially clash with an `overline` getter. I had the same problem with `glib::Param` while aligning getters with Rust styles in https://github.com/gtk-rs/gtk-rs/pull/211. See `get_flags` which was "rightfully" renamed as `flags` when there was already a `flags` constructor which lead me to rename all those with a `new` prefix, which I believe makes it clearer what the intent is.

**Comment by sdroege** [comment]
@fengalin Complicated. For "variant constructors" I saw in many libraries that they only use the name of the variant without `new_` as function name but that can indeed conflict with getters if the API is confusing enough like in paramspec (a `flags` variant and all pspecs also have flags... OTOH maybe there should be no `flags` constructor but it should be `ParamSpecFlags::new()` instead).

**Comment by fengalin** [comment]
Yes, the `Type::new` would probably be the way to go, or maybe a builder pattern.

**Comment by sdroege** [comment]
> Yes, the `Type::new` would probably be the way to go, or maybe a builder pattern.

They're actually different types that are returned there, so builder doesn't make too much sense. The main problem is that basically all other API wants a `ParamSpec` and not a `ParamSpecInt` (etc) so user code would have to cast or the `new` function would have to return something else than `Self`. Anyway, unrelated to this PR!

**Comment by fengalin** [comment]
I was talking about this PR too :) We could introduce an `AttributeBuilder` capable of building an `Attribute` with the expected variant. Ex. `AttriubuteBuilder::overline` (or longer `AttriubuteBuilder::build_overline`) wouldn't clash with an `Attribute` getter.

**Comment by sdroege** [comment]
That doesn't seem like an improvement in usability though :)

**Comment by fengalin** [comment]
Let's agree to disagree then :) I think we should come up with a general approach for this pattern so that it would be the same for all these situations in the organization though.

**Comment by sdroege** [comment]
Let's worry about that in a separate issue, can you create one? This one can go in as-is apart from the above questions then.

File: pango/src/attribute.rs (hunk 1 above)
Comment (GuillaumeGomez) [reply pango/src/attribute.rs:32]: My script is looking for `sys::` and not `ffi::`. ;)

Another run will be needed then!

Comment (GuillaumeGomez) [reply pango/src/attribute.rs:32]: Oh also! Would be nice to have the C equivalent for the types too.

Comment (sdroege) [reply pango/src/attribute.rs:32]: We don't use `sys::` anywhere since before you ran the script :P

Comment (GuillaumeGomez) [reply pango/src/attribute.rs:32]: Hum... Then the bug must be somewhere else.

File: pango/src/attribute.rs (hunk 2 above)
Comment (sophie-h) [reply pango/src/attribute.rs:103]: I would guess that none of them are nullable. I will remove Option from all of them.

**Comment by sdroege** [comment]
Thanks!

**Pull request merged**